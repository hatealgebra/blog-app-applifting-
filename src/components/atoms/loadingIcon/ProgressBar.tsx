import React from "react";

import { ProgressBar as Progress } from "react-loader-spinner";
import Theme from "../../particles/Theme";

const ProgressBar = () => {
  return (
    <Progress borderColor={Theme.color.border} barColor={Theme.color.primary} />
  );
};

export default ProgressBar;
