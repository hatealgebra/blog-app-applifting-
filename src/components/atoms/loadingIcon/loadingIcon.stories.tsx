import React from "react";
import LoadingCircles from "./Loading";
import ProgressBar from "./ProgressBar";

export const LoadingIcons = () => (
  <div style={{ display: "flex", gap: "30px" }}>
    <ProgressBar />
    <LoadingCircles />
  </div>
);

export default {
  title: "Atoms/Loading icons",
  subcomponent: { ProgressBar },
};
