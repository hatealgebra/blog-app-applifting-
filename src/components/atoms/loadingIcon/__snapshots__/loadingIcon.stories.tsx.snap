// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Loading icons LoadingIcons smoke-test 1`] = `
<div style="display: flex; gap: 30px;">
  <svg width="80"
       height="80"
       xmlns="http://www.w3.org/2000/svg"
       viewbox="0 0 100 100"
       preserveaspectratio="xMidYMid"
       class
       aria-label="progress-bar-loading"
       data-testid="progress-bar-svg"
       aria-busy="true"
       role="status"
  >
    <defs>
      <clipPath ng-attr-id="{{config.cpid}}"
                x="0"
                y="0"
                width="100"
                height="100"
                id="lds-progress-cpid-5009611b8a418"
      >
        <rect x="0"
              y="0"
              width="66.6667"
              height="100"
        >
          <animate attributename="width"
                   calcmode="linear"
                   values="0;100;100"
                   keytimes="0;0.5;1"
                   dur="1"
                   begin="0s"
                   repeatcount="indefinite"
          >
          </animate>
          <animate attributename="x"
                   calcmode="linear"
                   values="0;0;100"
                   keytimes="0;0.5;1"
                   dur="1"
                   begin="0s"
                   repeatcount="indefinite"
          >
          </animate>
        </rect>
      </clipPath>
    </defs>
    <path fill="none"
          ng-attr-stroke="{{config.c1}}"
          stroke-width="2.7928"
          d="M82,63H18c-7.2,0-13-5.8-13-13v0c0-7.2,5.8-13,13-13h64c7.2,0,13,5.8,13,13v0C95,57.2,89.2,63,82,63z"
          stroke="#DFDFDF"
    >
    </path>
    <path ng-attr-fill="{{config.c2}}"
          ng-attr-clip-path="url(#{{config.cpid}})"
          d="M81.3,58.7H18.7c-4.8,0-8.7-3.9-8.7-8.7v0c0-4.8,3.9-8.7,8.7-8.7h62.7c4.8,0,8.7,3.9,8.7,8.7v0C90,54.8,86.1,58.7,81.3,58.7z"
          fill="#007BFF"
          clip-path="url(#lds-progress-cpid-5009611b8a418)"
    >
    </path>
  </svg>
  <div data-testid="three-dots-loading"
       aria-label="three-dots-loading"
       aria-busy="true"
       role="status"
       style="display: flex;"
  >
    <svg width="80"
         height="80"
         viewbox="0 0 120 30"
         xmlns="http://www.w3.org/2000/svg"
         fill="#007BFF"
         data-testid="three-dots-svg"
    >
      <circle cx="15"
              cy="15"
              r="15"
      >
        <animate attributename="r"
                 from="15"
                 to="15"
                 begin="0s"
                 dur="0.8s"
                 values="15;9;15"
                 calcmode="linear"
                 repeatcount="indefinite"
        >
        </animate>
        <animate attributename="fill-opacity"
                 from="1"
                 to="1"
                 begin="0s"
                 dur="0.8s"
                 values="1;.5;1"
                 calcmode="linear"
                 repeatcount="indefinite"
        >
        </animate>
      </circle>
      <circle cx="60"
              cy="15"
              r="9"
              attributename="fill-opacity"
              from="1"
              to="0.3"
      >
        <animate attributename="r"
                 from="9"
                 to="9"
                 begin="0s"
                 dur="0.8s"
                 values="9;15;9"
                 calcmode="linear"
                 repeatcount="indefinite"
        >
        </animate>
        <animate attributename="fill-opacity"
                 from="0.5"
                 to="0.5"
                 begin="0s"
                 dur="0.8s"
                 values=".5;1;.5"
                 calcmode="linear"
                 repeatcount="indefinite"
        >
        </animate>
      </circle>
      <circle cx="105"
              cy="15"
              r="15"
      >
        <animate attributename="r"
                 from="15"
                 to="15"
                 begin="0s"
                 dur="0.8s"
                 values="15;9;15"
                 calcmode="linear"
                 repeatcount="indefinite"
        >
        </animate>
        <animate attributename="fill-opacity"
                 from="1"
                 to="1"
                 begin="0s"
                 dur="0.8s"
                 values="1;.5;1"
                 calcmode="linear"
                 repeatcount="indefinite"
        >
        </animate>
      </circle>
    </svg>
  </div>
</div>
`;
